lib/dry/core/constants.rb

```diff
@@ -91,6 +91,12 @@ def undefined.dup
         def undefined.clone
           self
         end
+
+        # @api public
+        def undefined.coalesce(*args)
+          args.each { |x| return x if !equal?(x) }
+          self
```

Comment:
Or in more functional way:

```suggestion
          args.find(-> { self }) { |x| !equal?(x) }
```

Reply:
this won't pass `expect(Undefined.coalesce).to be(Undefined)` and `expect(Undefined.coalesce(Undefined)).to be(Undefined)`

Reply:
hm, ok, I didn't know `find` accepts a default value

Reply:
@v-kolesnikov applied it in https://github.com/dry-rb/dry-core/commit/42b18ad4ae40faa9c68496671dbdb9e30fd4689b

Reply:
👍